service/tablet_allocator.cc

```diff
@@ -312,7 +312,7 @@ class load_balancer {
 
     struct skipped_candidate {
         tablet_replica replica;
-        global_tablet_id tablet;
+        global_tablet_id candidate;
```

Comment:
> migration_candidate -> logical candidate
> migration_candidate_info -> info about a migration candidate.

The new migration_candidate is not a full migration candidate. We move tablet replicas, but the new migration_candidate selects the tablet, not a particular replica. It makes sense when migration_candidate sits in the per-shard candidate list, since then it identifies tablet replica. But not when it sits in migration_candidate_info. 

I don't like the "_info" part because it's meaningless (what info?) as a distinctor. The names migration_candidate and migration_candidate_info are too similar. Perhaps a better name would be just "migration". migration_candidate_info selects the source and destination, so it's a full migration descriptor. Then the new "migration_candidate" could be salvaged.

Reply:
Though "migration_candidate" still wouldn't seem accurate, since it doesn't always select the replica. It depends on context. So "migration_tablet_set" seems still more accurate to me.

Reply:
Sure, will proceed that way.